pkg/generate/app/cmd/describe.go

```diff
@@ -171,11 +171,18 @@ func describeBuildPipelineWithImage(out io.Writer, ref app.ComponentReference, p
 				fmt.Fprintf(out, "      * The resulting image will be pushed to %s %q\n", to.Kind, to.Name)
 			}
 		}
-		if len(trackedImage) > 0 {
-			if noSource {
-				fmt.Fprintf(out, "      * Use 'start-build --from-dir=DIR' to trigger a new build\n")
-			} else {
+
+		if noSource {
+			// if we have no source, the user must always provide the source from the local dir(binary build)
+			fmt.Fprintf(out, "      * Use 'start-build --from-dir=DIR' to trigger a new build\n")
```

Comment:
we only mention --from-dir here, but then mention all 3 --from options on the next line - potentially confusion to newbies ?? .... perhaps have the above line not mention any of hte --from options, just the standard use star-build to trigger a new build, but then have the WARNING line below mention you have to pick from 1 of the 3 --from options?